hivemind/optim/collaborative.py

```diff
@@ -185,7 +191,14 @@ def is_alive(self) -> bool:
     def load_state_from_peers(self, **kwargs):
         """Attempt to fetch the newest collaboration state from other peers"""
         with self.lock_collaboration_state:
-            self.averager.load_state_from_peers(**kwargs)
+            while True:
+                try:
+                    self.averager.load_state_from_peers(timeout=self.load_state_timeout, **kwargs)
+                    break
+                except BaseException as e:
```

Comment:
```suggestion
                except Exception as e:
```

Otherwise, `KeyboardInterrupt` is ignored. Also, I doubt that we need to catch any exceptions not inherited from `Exception`.

Reply:
https://github.com/learning-at-home/hivemind/issues/372